browser/src/control/Control.JSDialogBuilder.js

```diff
@@ -1920,9 +1920,34 @@ L.Control.JSDialogBuilder = L.Control.extend({
 			};
 
 			text.addEventListener('click', clickFunction);
-			text.addEventListener('keypress', function onEvent(event) {
-				if (event.key === 'Enter')
+			text.addEventListener('keydown', function onEvent(event) {
+				var preventDef = false;
+				var listElements = $('#' + treeViewData.id + ' li');
+				var currIndex = parseInt(entry.row);
+				var treeLength = treeViewData.entries.length;
+				var spanElement = 'span.ui-treeview-cell';
+				if (event.key === 'Enter') {
 					clickFunction();
+					preventDef = true;
+				} else if (event.key === 'ArrowDown') {
+					if (currIndex === treeLength - 1)
+						listElements.eq(0).find(spanElement).focus();
+					else
+						listElements.eq(currIndex + 1).find(spanElement).focus();
+					preventDef = true;
+				} else if (event.key === 'ArrowUp') {
+					if (currIndex === 0)
+						listElements.eq(treeLength - 1).find(spanElement).focus();
+					else
+						listElements.eq(currIndex - 1).find(spanElement).focus();
+					preventDef = true;
+				} else if (treeViewData.id === 'mentionList' && event.key !== 'Tab' && event.key !== 'Shift') {
```

Comment:
hmm, I would prefer to keep widgets generic, not having any logic for particular widget instances for custom dialogs.
Can we do it different way?

For example pass custom keydown callback to the widget which will be called before original one and will return boolean which will indicate if event was consumed and doesn't have to be handled anymore.

Reply:
yep makes sense , thanks : )